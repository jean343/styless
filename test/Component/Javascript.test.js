import React from "react";
import renderer from "react-test-renderer";
import styled from "styled-components";
import "jest-styled-components";

test("Support javascript", () => {
  const Div = styled.div`
    @import "javascript.less";
    @color: green;
    color: @echoColor(@color);
    background: @primary-1;
    border: 1px solid @primary-2;
  `;
  expect(renderer.create(<Div />).toJSON()).toMatchSnapshot();
  expect(renderer.create(<Div color="red" />).toJSON()).toMatchSnapshot();
});
