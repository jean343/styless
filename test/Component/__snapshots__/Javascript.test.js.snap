// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support javascript 1`] = `
.c0 {
  color: green;
  background: #403d3c;
  border: 1px solid #403d3c;
}

<div
  className="c0"
/>
`;

exports[`Support javascript 2`] = `
.c0 {
  color: red;
  background: #403d3c;
  border: 1px solid #403d3c;
}

<div
  className="c0"
  color="red"
/>
`;
